src/robusta/core/sinks/robusta/rrm/rrm.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import threading
+import time
+from typing import List
+
+from robusta.core.model.env_vars import RRM_PERIOD_SEC, MANAGED_PROMETHEUS_ALERTS_ENABLED
+from robusta.core.sinks.robusta.rrm.account_resource_fetcher import AccountResourceFetcher
+from robusta.core.sinks.robusta.rrm.base_resource_manager import BaseResourceManager
+from robusta.core.sinks.robusta.rrm.prometheus_alert_resource_manager import \
+    PrometheusAlertResourceManager
+from robusta.core.sinks.robusta.rrm.types import ResourceKind
+
+
+# robusta resource management
+class RRM:
+    def __init__(self, dal: AccountResourceFetcher, cluster: str, account_id: str):
+        self.dal = dal
+        self.cluster = cluster
+        self.__sleep = RRM_PERIOD_SEC
+
+        self.__resource_managers: List[BaseResourceManager] = [
+            PrometheusAlertResourceManager(
+                resource_kind=ResourceKind.PrometheusAlert, cluster=self.cluster,
+                dal=dal)
+            ] if MANAGED_PROMETHEUS_ALERTS_ENABLED else []
+
+        self.__thread = threading.Thread(target=self.__thread_loop)
+        self.__thread.start()
+
+    def __thread_loop(self):
+        for res_man in self.__resource_managers:
+            try:
+                res_man.init_resources(updated_at=None)
+            except Exception:
+                self.__resource_managers.remove(res_man)
+                logging.warning(
+                    f"Unable to initialize the `{res_man.get_resource_kind()}` PrometheusRules CRD resources. Skipping it",
+                    exc_info=True)
+
+        while True:
+            try:
+                self.__periodic_loop()
+
+                time.sleep(self.__sleep)
+            except Exception as e:
+                logging.error(e)
+
+    def __periodic_loop(self):
+        for res_man in self.__resource_managers:
```

Comment:
I don't think we want to query each resource kind separately 
If we have 5 resource kinds, we will do 5 times the queries. (times the number of live clusters)
I think the logic should be:
```
Get all resources from saas
For each resource kind:
  - apply resource changes on the cluster
Update sync info to the saas  

```